README.md

```diff
@@ -71,3 +62,50 @@ services:
       - postgresql:/var/lib/postgresql
       - postgresql_data:/var/lib/postgresql/data
 ``` 
+
+# Configuration
+## Global configuration
+If you can define the properties globally (e.g. the pull request provider) within your SonarQube instance go to
+  [/admin/settings?category=pull+request](http://localhost:9000/admin/settings?category=pull+request) and set the
+   properties as admin.
+
+Set all other properties that you can define globally for all of your projects.
+
+Make sure `sonar.core.serverBaseURL` in SonarQube [/admin/settings](http://localhost:9000/admin/settings) is properly
+ set in order to for the links in the comment to work.
+
+## Project configuration
+Override global configuration on project level at [/project/settings?category=pull+request&id=PROJECT_KEY](http://localhost:9000/project/settings?category=pull+request&id=PROJECT_KEY) and set project specific
+ settings here.
+
+# Configuration of Bitbucket Server
+Set properties for Bitbucket Server either on Global configuration or Project configuration level as mentioned in the
+ Configuration section.
+
+## How to decorate the PR
+In order to decorate your short living branch (hotfixes, features, ...) pull-requests you need to analyze your long
+ living branches (master, development, release branches) first.
+ 
+You can configure patterns for long living branches in SonarQube.
+
+### Run analysis of long living branches
+  
+The analysis needs the following setting:
+`sonar.branch.name = branch_name (e.g master)`
+
+### Run analysis of the PR branch
+Carefully read the official SonarQube guide for [pull request decoration](https://docs.sonarqube.org/latest/analysis/pull-request/) 
+
+In there you'll find the following properties that need to be set.
+```
+sonar.pullrequest.key = pull_request_id (e.g. 100)
+sonar.pullrequest.branch = source_branch_name (e.g feature/TICKET-123)
+sonar.pullrequest.base = target_branch_name (e.g master)
+```
+
+:warning: There mustn't be any `sonar.branch` properties like `sonar.branch.name` available when you analyze a pull
+-request. As these properties indicate sonar that a branch is and no pull-request decoration is happening.
```

Comment:
This doesn't read quite right

```suggestion
:warning: There must not be any `sonar.branch` properties like `sonar.branch.name` arguments set when you analyze a pull-request. These properties indicate sonar that a branch is being analyzed rather than a pull-request so no pull-request decoration will be executed.
```